File: 프로그래머스/lv1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.js
function solution(s) {
    var answer = "";

    alpha = {"zero" : 0, "one" : 1, "two" : 2, "three" : 3, "four" : 4, "five" : 5,
            "six" : 6, "seven" : 7, "eight" : 8, "nine" : 9};

    let str = ""
    for (let i of s) {
        if (!isNaN(i)) {
            answer += i
        } else {
            str += i
            if (str in alpha) {
                answer += String(alpha[str])
                str = ""
            }
        }
    }

    return Number(answer);
}

File: 프로그래머스/lv1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.py
def solution(s):
    num_list = ['zero', 'one', 'two', 'three', 'four', 'five', 'six', 'seven', 'eight', 'nine']
    for idx, val in enumerate(num_list):
        s = str(idx).join(s.split(val))
    return int(s)